src/go_caw/main.go

```diff
@@ -0,0 +1,237 @@
+	flag_read         = flag.String("read", "", "Reads the caw thread starting at the given id")
+	flag_profile      = flag.Bool("profile", false, "Gets the user’s profile of following and followers")
+)
+
+// NOTE: these do not match the C++ Caw implementation order for enum in event_type.h
+// and enum CommandInput in caw.cc, because this is a "mixed" enumeration so that
+// there is no need for two separate enums in the same program.
+type EventType int
+
+const (
+	kRegisteruser EventType = iota
+	kCaw
+	kFollow
+	kRead
+	kProfile
+	kInvalidCommand
+	kHookAll
+	kUnhookAll
+	kReply
+)
+
+// extracts and validates command provided by global flags
+func getCommand() EventType {
+	var hook, unhook, registeruser, user, caw, reply, follow, read, profile bool = *flag_hookall, *flag_unhookall, *flag_registeruser != "", *flag_user != "", *flag_caw != "",
+		*flag_reply != "", *flag_follow != "", *flag_read != "", *flag_profile
+	if !registeruser && !user && !caw && !reply && !follow && !read &&
+		!profile {
+		if hook && !unhook {
+			return kHookAll // ./caw --hookall
+		} else if !hook && unhook {
+			return kUnhookAll // ./caw --unhookall
+		}
+	} else if !hook && !unhook {
+		if registeruser && !user && !caw && !reply && !follow && !read &&
+			!profile {
+			return kRegisteruser // ./caw --registeruser <username>
+		} else if !registeruser && user {
+			if caw && !follow && !read && !profile {
+				if caw && !reply {
+					return kCaw // ./caw --user <username> --caw <caw text>
+				} else if caw && reply {
+					return kReply // ./caw --user <username> --caw <caw text>
+					// --reply <reply caw id>
+				}
+			} else if !caw && follow && !read && !profile {
+				return kFollow // ./caw --user <username> --follow <username>
+			} else if !caw && !follow && read && !profile {
+				return kRead // ./caw --user <username> --read <caw id>
+			} else if !caw && !follow && !read && profile {
+				return kProfile // ./caw --user <username> --profile
+			}
+		}
+	}
+	return kInvalidCommand
+}
+
+// gets event request through serializing and packing payload
+func getEventRequest(request proto.Message, event EventType) *faz.EventRequest {
+	serialized_payload, err := ptypes.MarshalAny(request)
+	if err != nil {
+		panic(err)
+	}
+	event_request := &faz.EventRequest{
+		EventType: int32(event),
+		Payload:   serialized_payload,
+	}
+	return event_request
+}
+
+func main() {
+	flag.Parse()
+
+	// GRPC
+	var conn *grpc.ClientConn
+	conn, err := grpc.Dial("0.0.0.0:50000", grpc.WithInsecure())
+	if err != nil {
+		fmt.Println("sorry, could not connect to Faz")
+		log.Fatalf("did not connect: %s", err)
+		conn.Close()
+		os.Exit(1)
+	}
+	client := faz.NewFazServiceClient(conn)
+	command := getCommand()
+	event_map := map[EventType]string{
+		kRegisteruser: "registeruser",
+		kCaw:          "caw",
+		kFollow:       "follow",
+		kRead:         "read",
+		kProfile:      "profile",
+	}
+
+	if command == kInvalidCommand {
```

Comment:
Program should terminate with exit code 1 here as well

Reply:
good catch!